caom2pipe/caom2pipe/caom_composable.py

```diff
@@ -0,0 +1,148 @@
+#  arising from the use of the          accessoire ou fortuit, résultant
+#  software.  Neither the name          de l'utilisation du logiciel. Ni
+#  of the National Research             le nom du Conseil National de
+#  Council of Canada nor the            Recherches du Canada ni les noms
+#  names of its contributors may        de ses  participants ne peuvent
+#  be used to endorse or promote        être utilisés pour approuver ou
+#  products derived from this           promouvoir les produits dérivés
+#  software without specific prior      de ce logiciel sans autorisation
+#  written permission.                  préalable et particulière
+#                                       par écrit.
+#
+#  This file is part of the             Ce fichier fait partie du projet
+#  OpenCADC project.                    OpenCADC.
+#
+#  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+#  you can redistribute it and/or       vous pouvez le redistribuer ou le
+#  modify it under the terms of         modifier suivant les termes de
+#  the GNU Affero General Public        la “GNU Affero General Public
+#  License as published by the          License” telle que publiée
+#  Free Software Foundation,            par la Free Software Foundation
+#  either version 3 of the              : soit la version 3 de cette
+#  License, or (at your option)         licence, soit (à votre gré)
+#  any later version.                   toute version ultérieure.
+#
+#  OpenCADC is distributed in the       OpenCADC est distribué
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  $Revision: 4 $
+#
+# ***********************************************************************
+#
+
+import logging
+
+from caom2 import CompositeObservation, Algorithm, TypedSet
+from caom2 import ObservationURI, PlaneURI
+
+from caom2pipe import execute_composable as ec
+from caom2pipe import manage_composable as mc
+
+__all__ = ['change_to_composite', 'update_plane_provenance',
+           'update_observation_members']
+
+
+def change_to_composite(observation):
```

Comment:
This bit of code is used by both OMM and Gemini. It contains three bits of knowledge: the acceptable algorithm name, the data members from a Simple Observation that are also available in a Composite Observation, and the order in which those data members need to be provided to the Composite Observation constructor. IMHO those three bits of information justify a method, and when 2.4 comes out with the Derived Observation instead, it will make it easier to fix all the applications.

Do you have a recommendation for where it should go in caom2utils?

Reply:
There might be no use case for this function if observations can be saved as composite with 0 members...

Reply:
Moved to omm2caom2, where it is still needed, because there are SimpleObservation records in the database that need to be CompositeObservation instances.